eq-author/src/App/publish/PublishPage.js

```diff
@@ -82,61 +125,84 @@ const PublishPage = ({ match, history }) => {
-                  value={inputs.surveyId}
+        <StyledPanel>
+          <Field>
+            <Label>Themes</Label>
+            <Caption>Select themes to be used for this survey</Caption>
+            <ThemeContainer>
+              {themes.map(theme => (
+                <LabeledCheckbox
+                  id={`${theme}-selector`}
+                  key={`${theme}-selector`}
+                  handleChange={handleThemeSelect}
+                  label={theme}
                 />
-              </Shadow>
-            </Field>
-          </AlignedColumn>
+              ))}
+            </ThemeContainer>
+          </Field>
           <Separator />
-          <AlignedColumn cols={6}>
-            <Field>
-              <Label htmlFor="formType">Form type</Label>
-              <Caption>Enter form type</Caption>
-              <Shadow>
-                <Input
-                  id="formType"
-                  onChange={handleInputChange}
-                  value={inputs.formType}
-                />
-              </Shadow>
-            </Field>
-          </AlignedColumn>
+          <Field>
+            <Label htmlFor="surveyId">Survey ID</Label>
+            <Caption>Enter survey ID</Caption>
+            <Shadow>
+              <Input
+                id="surveyId"
+                onChange={e => setSurveyId(e.value)}
+                value={surveyId}
+              />
+            </Shadow>
+          </Field>
           <Separator />
-          <AlignedColumn>
-            <InformationPanel maxWidth="50%">
-              No further changes can be made to the questionnaire after it has
-              been submitted for approval
-            </InformationPanel>
-          </AlignedColumn>
+          {!isEmpty(formTypes) && (
+            <>
+              <Field>
+                <Label>Form type</Label>
+                <Caption>Enter relevant form types for selected themes</Caption>
+                <ThemeInputs>
+                  {map(formTypes, (value, key) => (
+                    <Shadow key={`${key}-entry`}>
+                      <Label htmlFor={key}>{key}</Label>
+                      <Input
+                        id={`${key}`}
+                        onChange={handleInputChange}
+                        value={value}
+                        data-test={`${key}-input`}
+                      />
+                    </Shadow>
+                  ))}
+                </ThemeInputs>
+              </Field>
+              <Separator />
+            </>
+          )}
+          <InformationPanel>
+            No further changes can be made to the questionnaire after it has
+            been submitted for approval
+          </InformationPanel>
           <Button
             type="submit"
             variant="primary"
-            disabled={!(inputs.surveyId && inputs.formType)}
+            disabled={
+              !surveyId ||
+              isEmpty(formTypes) ||
+              Object.keys(formTypes).some(k => !formTypes[k])
```

Comment:
Disabled here can just be `disabled={!surveyId || some(formTypes, isEmpty)}`

Reply:
Needs the isEmpty(formTypes) just is case no formtypes but the other simplification is good.